File: Programming Tasks/C/Q30 (selection_sort).c
#include<stdio.h>
void f(int a[],int n){
 int i,j,min_i,t;
 for(i=0;i<n-1;i++){
  min_i=i;
  for(j=i+1;j<n;j++)
   if(a[j]<a[min_i]) min_i=j;
  t=a[i];a[i]=a[min_i];a[min_i]=t;
 }
 for(i=0;i<n;i++)printf("%d ",a[i]);
}
int main(){
 int a[]={4,2,3,1};
 f(a,4);
 return 0;
}

File: Programming Tasks/Python/Q30 (selection_sort).py
def f(a):
 n=len(a)
 for i in range(n):
  min_i=i
  for j in range(i+1,n):
   if a[j]<a[min_i]: min_i=j
  a[i],a[min_i]=a[min_i],a[i]
 print(a)

f([4,2,3,1])
